test/reservation_test.rb

```diff
@@ -0,0 +1,116 @@
+require_relative "test_helper.rb"
+
+describe "reservation class" do
+  before do
+    @room = Stayappy::Room.new(room_num: 1, cost: 200)
+    @check_in = Date.new(2020, 3, 10)
+    @check_out = Date.new(2020, 3, 11)
+    @reservation = Stayappy::Reservation.new(@room, @check_in, @check_out)
+  end
+
+  describe "Initialize" do
+    it "can create a reservation" do
+      expect(@reservation).must_be_instance_of Stayappy::Reservation
+    end
+
+    it "can set the room" do
+      expect(@reservation.room).must_equal @room
+    end
+
+    it "can set the check_in and check_out duration" do
+      expect(@reservation.check_in).must_equal @check_in
+      expect(@reservation.check_out).must_equal @check_out
+    end
+
+    it "can determine the stay correctly" do
+      expect(@reservation.stay).must_equal @check_in..@check_out
+    end
+
+    it "can raises an error if the check_in date is after the check_out date" do
+      expect { Stayappy::Reservation.new(@room, @check_out, @check_in) }.must_raise ArgumentError
+    end
+  end
+
+  describe "in_range method" do
```

Comment:
Nice work on these test. It looks like there's one line of code that's not tested that may cover a couple tests cases. Review the test cases from the design scaffold. 

```
describe "overlap?" do
    before do
      start_date = Date.new(2017, 01, 01)
      end_date = start_date + 3

      @range = Hotel::DateRange.new(start_date, end_date)
    end

    it "returns true for the same range" do
      start_date = @range.start_date
      end_date = @range.end_date
      test_range = Hotel::DateRange.new(start_date, end_date)

      expect(@range.overlap?(test_range)).must_equal true
    end

    xit "returns true for a contained range" do
    end

    xit "returns true for a range that overlaps in front" do
    end

    xit "returns true for a range that overlaps in the back" do
    end

    xit "returns true for a containing range" do
    end

    xit "returns false for a range starting on the end_date date" do
    end

    xit "returns false for a range ending on the start_date date" do
    end

    xit "returns false for a range completely before" do
    end

    xit "returns false for a date completely after" do
    end
  end
```